src/publish.ts

```diff
@@ -0,0 +1,135 @@
+import { isMapDocumentNode, isProfileDocumentNode } from '@superfaceai/ast';
+import { isProviderJson } from '@superfaceai/one-sdk';
+import { parseMap, parseProfile, Source } from '@superfaceai/parser';
+import { ServiceApiError, ServiceClient } from '@superfaceai/service-client';
+
+import {
+  exists,
+  EXTENSIONS,
+  localMaps,
+  localProfiles,
+  localProviders,
+  PrintFn,
+  readFile,
+} from './util';
+
+export type PublishOptions = {
+  print?: PrintFn;
+  baseUrl?: string;
+  refreshToken?: string;
+};
+
+export async function publish(
+  path: string,
+  options?: PublishOptions
+): Promise<void> {
+  if (!(await exists(path))) {
+    throw new Error('Path does not exist');
+  }
+
+  const client = new ServiceClient({
+    baseUrl: options?.baseUrl,
+    refreshToken: options?.refreshToken,
+  });
+
+  const file = await readFile(path);
+
+  if (path.endsWith(EXTENSIONS.provider)) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const parsedFile = JSON.parse(file);
+    if (isProviderJson(parsedFile)) {
```

Comment:
Maybe we could use `parseProviderJson` instead. 

Parse throws helpful errors :)

Reply:
I don't think it is necessary. Publish is supposed to be used in CI/CD only and get to this point it must have passing tests. And in case someone calls it (for whatever reason) I am ok to keep it on server to decide if it is parsable or not.